### DISCUSSION TITLE
Question: Is it possible to change the domain on which the worker is registered?

### DISCUSSION THREAD

**Discussion by gCardinal** [mswjs/msw#801] [Q&A] [answered]
I am running my app in a Docker container and routing to it via a global a local nginx container and dnsmasq. In short, the app runs on the url, for example, `http://crew.shadowshifts.local.majisti.com/`. However, it seems like there is no way to have the service worker register on this URL, which causes issues I have wrongly though solved in #796.

When I run via a localhost, the service worker registers properly. 

[Non-locahost URL](https://i.imgur.com/DCe4B9g.png)
[Localhost URL](https://i.imgur.com/9R0qJRu.png)

**Comment by kettanaito** [comment]
Hey, @gCardinal. 

There are multiple [service worker options](https://mswjs.io/docs/api/setup-worker/start#serviceworker) you can configure when calling `worker.start()`. It looks like in your case the `serviceWorker.url` may be useful:

```js
worker.start({
  serviceWorker: {
    url: IS_DOCKER
      ? 'http://crew.shadowshifts.local.majisti.com/mockServiceWorker.js'
      : '/mockServiceWorker.js'
  }
})
```

Note that the worker script must be served from the same domain that your application where you intend to use MSW.

**Reply by gCardinal** [reply]
I had seen the `url` option, but it didn't work. Turns out, if you're not running from localhost, you **_need_** to be using HTTPS or the service worker won't be registered in the browser.

Your answer did point me in the right direction, thank you.

**Reply by kettanaito** [reply]
My bad, I didn't realize we weren't in a localhost context anymore. Glad you've found that requirement!

**Comment by gCardinal** [comment, marked as answer]
Service workers require the page to be secure (HTTPS) or the browser will refuse to register the service worker (even if `msw` logs that the service worker is registered).

Using HTTPS on local (via [mkcert](https://github.com/FiloSottile/mkcert) and a nginx reverse proxy) solve the issue for me.
